### PR TITLE
Add option to manage only GID instead of all groups.

### PR DESCRIPTION
**Pull request by bbriggs**

### CODE REVIEW

**Comment by bbriggs** [comment]
So I'm not able to kick off a Travis build and I'm not sure why.

File: manifests/init.pp [hunk 1]
```diff
@@ -61,13 +61,24 @@
     if ($uid) {
       validate_integer($uid)
     }
-    if ($manage_groups) {
-      $managed_groups = [$groups, $gid]
+
+    case $manage_groups {
```
Comment (rnelson0) [inline manifests/init.pp]: Leave the `if` around this (see below)

File: manifests/init.pp [hunk 2]
```diff
@@ -61,13 +61,24 @@
...
+      true:  {
+        $managed_groups = [$groups, $gid]
+      }
+      'primary','gid':  {
+        $managed_groups = [$gid]
+      }
+      default: { }
```
Comment (rnelson0) [inline manifests/init.pp]: `false` and `'disabled'` should create an empty array, and other values should generate an error with `fail()`.

File: spec/defines/init_spec.rb [hunk 3]
```diff
@@ -42,6 +42,26 @@
...
+
+    it { is_expected.to create_user('rnelson0').with({
+      :comment          => 'Rob Nelson',
+      :groups           => ['group1', 'group2'],
+    }) }
+    it { is_expected.to create_group('rnelson0') }
+    it { is_expected.not_to create_group('group1') }
+    it { is_expected.not_to create_group('group2') }
+  end
+
```
Comment (rnelson0) [inline spec/defines/init_spec.rb]: Add a test for `disabled|false` and `garbage` values, testing for no managed groups or an error code.

File: manifests/init.pp [hunk 4]
```diff
@@ -61,13 +61,24 @@
     if ($uid) {
       validate_integer($uid)
     }
-    if ($manage_groups) {
-      $managed_groups = [$groups, $gid]
+
+    case $manage_groups {
+      true:  {
```
Comment (rnelson0) [inline manifests/init.pp]: `enabled` is a good alternative, as below.

**Comment by rnelson0** [comment]
I added some comments but I am going to close this. Fork the repo and push your branch to your fork, then open a PR.

**Pull request closed without merging**